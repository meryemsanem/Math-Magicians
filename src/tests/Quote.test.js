import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import QuotePage from '../components/Quote';

test('renders the QuotePage component', () => {
  render(
    <MemoryRouter>
      <QuotePage />
    </MemoryRouter>,
  );

  const navbarElement = screen.getByRole('navigation');
  expect(navbarElement).toBeInTheDocument();

  const quotesElement = screen.getByRole('main');
  expect(quotesElement).toBeInTheDocument();

  const componentTree = renderer.create(
    <MemoryRouter>
      <QuotePage />
    </MemoryRouter>,
  );
  expect(componentTree.toJSON()).toMatchSnapshot();
});
