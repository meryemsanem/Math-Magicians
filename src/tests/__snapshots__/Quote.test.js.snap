// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the QuotePage component 1`] = `
<div>
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <div />
  </main>
</div>
`;
